w3-aggregation.md

```diff
@@ -0,0 +1,396 @@
+
+### Storefront can query state of the aggregate deals
+
+Storefront users MAY want to check details about deals from the content they previously stored. These deals will change over time as they get renewed. Therefore, Storefront should invoke `aggregate/get` capability to gather information about given aggregate identifier.
+
+```mermaid
+sequenceDiagram
+participant Storefront as 🌐<br/><br/>did:web:web3.storage
+    participant Authority as 🌐<br/><br/>did:web:spade.storage
+
+    Storefront->>Authority: invoke `aggregate/get`
+```
+
+## Capabilities
+
+This section describes the capabilities that form the w3 aggregation protocol, along with the details relevant for invoking capabilities with a service provider.
+
+In this document, we will be looking at `spade.storage` as an implementer of the `aggregate/*` and `offer/*` protocol.
+
+### `aggregate/offer`
+
+A Storefront principal can invoke a capabilty to submit an aggregate ready for offers.
+
+```iplsch
+type AggregateOffer struct {
+  with StorefrontDID
+  nb AggregateOfferDetail
+}
+
+type AggregateOfferDetail struct {
+  offer OfferCBOR
+  commitmentProof Proof
+}
+
+type OfferCBOR any
+type Proof any
+
+type struct OfferDetails {
+    size Int
+    link Link
+    src [URL]
+    commitmentProof Proof
+}
+
+type Offer [OfferDetails]
+```
+
+> `did:web:web3.storage` invokes capability from `did:web:spade.storage`
+
+```json
+{
+  "iss": "did:web:web3.storage",
+  "aud": "did:web:spade.storage",
+  "att": [{
+    "with": "did:web:spade.storage",
+    "can": "aggregate/offer",
+    "nb": {
+      "offer": { "/": "bafy...many-cars" }, /* dag-cbor CID */
+      "commitmentProof": { "/": "commitment...cars-proof" } /* commitment proof */
+    }
+  }],
+  "prf": [],
+  "sig": "..."
+}
+```
+
+Invoking `aggregate/offer` capability submits an offer to a broker service to arrange a Filecoin deals. The `nb.offer` represents a "Ferry" aggregate offer that is ready for a Filecoin deal. Its value is the DAG-CBOR CID that refers to a "Ferry" offer. It encodes a dag-cbor block with an array of entries representing all the CAR files to include in the aggregated deal. This block MUST be included in CAR file that transports the invocation. Its format is:
+
+```json
+/* decoded offers block as OfferDetails type */
+[ {
+    "link": { "/": "bag...file0" }, /* CAR CID */
+    "size": 110101,
+    "commitmentProof": { "/": "commitment...car0" }, /* COMMP CID */
+    "src": ["https://w3s.link/ipfs/bag...file0"]
+  },
+  {
+    /* ... */
+  }
+]
+```
+
+Each entry of the decoded offers block, has all the necessary information for a Storage Provider to fetch and store a CAR file. The `link` field has the CAR CID, while the `commitmentProof` field has the required `proof` bytes by Storage Providers (for example, `commP`). The `src` field of each piece MUST be set to a (alphabetically sorted) list of URLs from which it can be fetched. The `size` field MUST be set to the byte size of the CAR file.
```

Comment:
Do we want to specify that the `src` field is optional (allowing it to be provided when the deal is signed)?